sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala

```diff
@@ -525,7 +525,7 @@ private[sql] object CatalogV2Util {
     }
 
     if (isDefaultColumn) {
-      val e = analyze(f, EXISTS_DEFAULT_COLUMN_METADATA_KEY)
+      val e = analyze(f, statementType = "Column conversion", EXISTS_DEFAULT_COLUMN_METADATA_KEY)
```

Comment:
Statement type is used only for debugging/logging, the main purpose of it is to be present in exception message being thrown.

Reply:
I am not sure whether `columns` method of `Table` is called only on creation

Reply:
At any rate, this looks like the right fix!

Reply:
Thanks, I added more changes.